### ISSUE TITLE
Node issue after following install steps

### ISSUE THREAD

**Issue by derpynirvash** [calzoneman/sync#177]
After following the CentOs install steps on this page, https://github.com/calzoneman/sync/wiki/Installing
And getting around a few hicups with Node, I have gotten stuck with this error when trying to launch cytube. And an unsure how to get around it. Any help would be very appreciated.

Starting CyTube v1.9.5

/usr/local/lib/node_modules/mysql-libmysqlclient/lib/mysql-libmysqlclient.js:208
MysqlConnectionQueued.prototype = new bindings.MysqlConnection();
                                              ^
TypeError: Cannot read property 'MysqlConnection' of null
    at Object.<anonymous> (/usr/local/lib/node_modules/mysql-libmysqlclient/lib/mysql-libmysqlclient.js:208:47)
    at Module._compile (module.js:456:26)
    at Object.Module._extensions..js (module.js:474:10)
    at Module.load (module.js:356:32)
    at Function.Module._load (module.js:312:12)
    at Module.require (module.js:364:17)
    at require (module.js:380:17)
    at Object.<anonymous> (/usr/src/sync/auth.js:12:13)
    at Module._compile (module.js:456:26)
    at Object.Module._extensions..js (module.js:474:10)

**Comment by calzoneman** [comment]
Are you sure you performed this step from the installation guide?

> Install MySQL+tools: `yum install mysql-server mysql php-mysql mysql-devel mysql-libs`

The error you're getting suggests that the node module `mysql-libmysqlclient` can't find the shared library that backs it.

**Comment by derpynirvash** [comment]
I am sure I did that step.
Also for the sake of troubleshooting, I rolled the server back to before I
added Node/synch and tried installing again, to get the same error.

I am going to try installing sync on a clean install of centos later today
and will let you know what I get.

On Sun, Jun 23, 2013 at 9:19 AM, Calvin Montgomery <notifications@github.com

**Comment by nuclearace** [comment]
I had that problem as well setting up on my raspberry pi. You'll need to make a symbolic link pointing to the shared library.

**Comment by calzoneman** [comment]
What are you setting this up on?  I had v1.9.5 running on a CentOS server (OpenVZ) and it worked fine.  I didn't follow the exact install steps from the wiki though (I installed to a home directory rather than `/usr/src`

**Comment by derpynirvash** [comment]
Sorry for the very late reply, I had a business trip last week and just got home. The server I was using to test these was CentOS 6.3, one was previously configured to handle backups and I also tried it on a fresh install. I however was able to test it on a server that was configured as a webserver (centOS 6.4) and it worked there (Possible version conflict?).
However that server wouldn't be ideal to run the client side on. Which also brings the question, would the client side and server side be able to run on different machines?

And @nuclearace Where would the shared library be?

**Comment by calzoneman** [comment]
I'm not sure what you mean... what I refer to as "the clientside" is just HTML and Javascript that gets served to a web browser.  It works on pretty much everything I've tried, although there is some weird behaviour in Internet Explorer.

**Comment by derpynirvash** [comment]
Sorry if that wasn't clear, I was just wondering, as a side note, if it
would be possible to run the 'server' on one machine and have the web
portion hosted off a different machine. (Assuming all the ports and such
are open on both machines)

**Comment by calzoneman** [comment]
It is possible, but I don't really recommend it.  The HTML/CSS/Javascript can be served from an external server as long as you configure `IO_URL` and `WEB_URL` to point to the CyTube server.  However, some functions, including logins and account management will depend on the built-in webserver anyways, so you're not going to save many resources by doing so.

It also requires some `.htaccess` trickery to get the URLs to work properly (I'll have to go find it in another ticket, because someone else asked about Apache).

I used to serve my web content from Apache, but I stopped doing that because it was cumbersome to maintain in my opinion.  My current setup is using nginx with the tcp_proxy module to route traffic on port 80 of my desired hostname to the CyTube instance running on a different port.
